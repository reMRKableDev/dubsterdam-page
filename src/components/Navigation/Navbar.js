import React from "react";
import NavbarButton from "./NavbarButton";
import { StyledNav, StyledThemeButtonContainer } from "./NavbarStyles";

const Navbar = ({ theme, handleThemeToggle }) => {
  return (
    <StyledNav data-testid="app-navbar">
      <StyledThemeButtonContainer>
        <NavbarButton theme={theme} handleThemeToggle={handleThemeToggle} />
      </StyledThemeButtonContainer>
    </StyledNav>
  );
};

export default Navbar;
