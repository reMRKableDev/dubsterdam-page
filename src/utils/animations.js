import { keyframes } from "styled-components";
import { bounce, fadeInUp, fadeInDown, shake, merge } from "react-animations";

export const bounceAnimation = keyframes`${bounce}`;
export const fadeInUpAnimation = keyframes`${fadeInUp}`;
export const fadeInDownAnimation = keyframes`${fadeInDown}`;
export const shakeAnimation = keyframes`${shake}`;
export const fadeInUpShakeAnimation = keyframes`${merge(fadeInUp, shake)}`;
