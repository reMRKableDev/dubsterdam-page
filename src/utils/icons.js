import { Mail, Github, Linkedin, Database } from "@styled-icons/feather";
import {
  Css3,
  Html5,
  ReactLogo,
  Nodejs,
  Javascript,
} from "@styled-icons/boxicons-logos";

export const mailIcon = Mail;
export const gitHubIcon = Github;
export const linkedInIcon = Linkedin;
export const databaseIcon = Database;
export const cssIcon = Css3;
export const htmlIcon = Html5;
export const reactIcon = ReactLogo;
export const nodeIcon = Nodejs;
export const javascriptIcon = Javascript;
