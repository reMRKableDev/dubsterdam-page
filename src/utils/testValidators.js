export const validateNotEmpty = (received) => {
  expect(received).not.toBeNull();
  expect(received).toBeTruthy();
};

export const validateDarkThemeMode = (received) => {
  expect(received).not.toMatch(/light/i);
  expect(received).toMatch(/dark/i);
};

export const validateLightThemeMode = (received) => {
  expect(received).not.toMatch(/dark/i);
  expect(received).toMatch(/light/i);
};
