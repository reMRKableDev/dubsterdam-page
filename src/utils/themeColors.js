import theme from "styled-theming";

export const bgColor = theme("mode", {
  light: "#fafafa",
  dark: "#222",
});

export const footerColor = theme("mode", {
  light: "#222",
  dark: "#fafafa",
});

export const textColor = theme("mode", {
  light: "#000",
  dark: "#fff",
});

export const buttonBgColor = theme("mode", {
  light: "#222",
  dark: "#eee",
});

export const buttonTextColor = theme("mode", {
  light: "#eee",
  dark: "#222",
});

export const brandColor = theme("mode", {
  light: "#ed6c6d",
  dark: "#ed6c6d",
});

export const textAccentColor = theme("mode", {
  light: "#555",
  dark: "#fff",
});
