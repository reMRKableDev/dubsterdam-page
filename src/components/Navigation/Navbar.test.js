import React from "react";
import { shallow } from "enzyme";
import Navbar from "./Navbar";
import { validateMatchSnapshot } from "../../utils/testValidators";

describe("Navbar unit test", () => {
  it("should validate match snapshot to Navbar component with theme 'Light Mode ðŸŒ•'", () => {
    const dummyTheme = {
      mode: "dark",
    };
    const wrapper = shallow(<Navbar theme={dummyTheme} />);

    validateMatchSnapshot(wrapper);
  });

  it("should validate match snapshot to Navbar component with theme 'Dark Mode ðŸŒ‘'", () => {
    const dummyTheme = {
      mode: "light",
    };
    const wrapper = shallow(<Navbar theme={dummyTheme} />);

    validateMatchSnapshot(wrapper);
  });
});
