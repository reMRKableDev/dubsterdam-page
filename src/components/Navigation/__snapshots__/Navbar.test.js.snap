// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar unit test should validate match snapshot to Navbar component with theme 'Dark Mode ðŸŒ‘' 1`] = `
<styled.nav>
  <styled.div>
    <styled.button>
      Dark Mode ðŸŒ‘
    </styled.button>
  </styled.div>
</styled.nav>
`;

exports[`Navbar unit test should validate match snapshot to Navbar component with theme 'Light Mode ðŸŒ•' 1`] = `
<styled.nav>
  <styled.div>
    <styled.button>
      Light Mode ðŸŒ•
    </styled.button>
  </styled.div>
</styled.nav>
`;
